### PR TITLE
View manager props

### PR DESCRIPTION
**Pull request by erksch**
This MR adds the capability to annotate Compose function parameters with @ReactNativeProp to make them settable from React Native.

### Android

For Android, supporting props is straight-forward. Each prop gets a setter annotated with `@ReactProp` in the generated `SimpleViewManager<ComposeView>` subclass (see [React Native docs on this](https://reactnative.dev/docs/native-components-android#3-expose-view-property-setters-using-reactprop-or-reactpropgroup-annotation)). The setter emits a value into shared flow which is passed into the compose function.

For callback parameters of the compose function we generate implementations of the callback according to the [React Native Android view manager event docs](https://reactnative.dev/docs/native-components-android#events).

By generating all this into the view manager everything works out of the box.

### iOS

On iOS, this is much more tricky. 

Firstly, React Native requires the that you add `RCT_EXPORT_VIEW_PROPERTY` on your `RCTViewManager` implementation which [under hood creates a class method `propConfig_<name>`](https://github.com/facebook/react-native/blob/7cdb87eb192bc932ddb07a723aa3292c08606bf8/packages/react-native/React/Views/RCTViewManager.h#L67). [There is no way to write a Kotlin class in ios source sets so that it has class methods](https://youtrack.jetbrains.com/issue/KT-71364/Kotlin-companion-object-methods-are-not-mapped-to-Objective-C-class-methods) so we need to get rid of the Kotlin subclassing of `RCTViewManager` (and remove the interop) and instead generate the subclass in Objective-C code so we can set the class methods properly.

Secondly, when React Native picked which props exist via the `RCT_EXPORT_VIEW_PROPERTY` methods it calls the method **directly the `UIView`** returned by the view manager (you can circumvent this using `RCT_CUSTOM_VIEW_PROPERTY` but it is true at least for callback props). Now we run into the second Kotlin / Objective-C interop problem: We can not create a class in Kotlin ios source sets that subclasses an Objective-C type (`UIView` in this case) and add methods to it that are callable from Objective-C. So we also need to generate the `UIView` subclass as Objective-C code and add the Compose view as a subview to it.

So on iOS we have 3 components:
- The generated `RCTViewManager` subclass which configures the prop configs
- The generated `UIView` subclass which provides properties for callback props
- A generated `ReactNativeIOSViewManager` which provides the glue code between the Objective-C `RCTViewManager` and `UIView` and the Kotlin Compose view.

We generate all Objecitve-C code in one header and one implementation file that the library user has to include in their XCode project.

### Future work
- Generate TypeScript interface and code for the native view

### CODE REVIEW

File: example/android/shared/src/commonMain/kotlin/com/myrnproject/shared/MyComposeView.kt [hunk 1]
```diff
@@ -1,15 +1,44 @@
...
 @Composable
 @ReactNativeViewManager("MyComposeView")
-internal fun MyComposeView(persistentConfig: PersistentConfig) {
+internal fun MyComposeView(
+    @ReactNativeProp
+    message: Flow<String>,
+    @ReactNativeProp
+    textFieldValue: Flow<String>,
+    @ReactNativeProp
+    onTextFieldValueChange: (String) -> Unit,
```
Comment (Legion2) [inline example/android/shared/src/commonMain/kotlin/com/myrnproject/shared/MyComposeView.kt:22]: Should this be suspend functions? Or are the calls blocking?

File: kotlin/reakt-native-toolkit/src/iosMain/kotlin/de/voize/reaktnativetoolkit/util/ReactNativeIOSViewManager.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package de.voize.reaktnativetoolkit.util
+
+import platform.UIKit.UIView
+
+open class ReactNativeIOSViewManager {
```
Comment (Legion2) [inline kotlin/reakt-native-toolkit/src/iosMain/kotlin/de/voize/reaktnativetoolkit/util/ReactNativeIOSViewManager.kt:5]: What is this for, please add a comment

File: example/src/components/nativeViews.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+const MyComposeViewNative = requireNativeComponent<any>('MyComposeView');
+
+export const MyComposeView = ({
+  onTextFieldValueChange,
+  onButtonPress,
+  ...props
+}: any) => {
+  const nativeOnTextFieldValueChange = useCallback(
+    (event: any) => {
+      onTextFieldValueChange?.(...event.nativeEvent.args);
```
Comment (Legion2) [inline example/src/components/nativeViews.tsx:13]: This should be generated

**Comment by erksch** [comment]
https://github.com/user-attachments/assets/c454320c-ddb1-4108-ba61-aca606d64fe3

File: example/android/shared/src/commonMain/kotlin/com/myrnproject/shared/MyComposeView.kt (hunk 1 above)
Comment (erksch) [reply example/android/shared/src/commonMain/kotlin/com/myrnproject/shared/MyComposeView.kt:22]: nope this not blocking

File: kotlin/reakt-native-toolkit/src/iosMain/kotlin/de/voize/reaktnativetoolkit/util/ReactNativeIOSViewManager.kt (hunk 2 above)
Comment (erksch) [reply kotlin/reakt-native-toolkit/src/iosMain/kotlin/de/voize/reaktnativetoolkit/util/ReactNativeIOSViewManager.kt]: done

File: example/android/shared/src/commonMain/kotlin/com/myrnproject/shared/MyComposeView.kt [hunk 4]
```diff
@@ -1,15 +1,65 @@
...
+    Column {
+        Text("Hello from Compose, ${name ?: "unknown"}!")
+
+        listOf(
+            "message" to messageValue,
+            "boolProp" to boolValue,
+            "intProp" to intValue,
+            "doubleProp" to doubleValue,
+            "floatProp" to floatValue,
+        ).forEach { (propName, propValue) ->
```
Comment (Legion2) [inline example/android/shared/src/commonMain/kotlin/com/myrnproject/shared/MyComposeView.kt:52]: Add test for data class

File: example/android/shared/src/commonMain/kotlin/com/myrnproject/shared/MyComposeView.kt [hunk 5]
```diff
@@ -1,15 +1,65 @@
...
+    @ReactNativeProp
+    intProp: Flow<Int>,
+    @ReactNativeProp
+    doubleProp: Flow<Double>,
+    @ReactNativeProp
+    floatProp: Flow<Float>,
+    @ReactNativeProp
+    onTextFieldValueChange: (String) -> Unit,
+    @ReactNativeProp
+    onButtonPress: () -> Unit,
```
Comment (Legion2) [inline example/android/shared/src/commonMain/kotlin/com/myrnproject/shared/MyComposeView.kt:32]: Test callbacks with data classes, multiple params, return types and exceptions.

**Review by Legion2 — CHANGES_REQUESTED** [review]

File: example/android/shared/src/commonMain/kotlin/com/myrnproject/shared/MyComposeView.kt (hunk 4 above)
Comment (erksch) [reply example/android/shared/src/commonMain/kotlin/com/myrnproject/shared/MyComposeView.kt:75]: done

File: example/android/shared/src/commonMain/kotlin/com/myrnproject/shared/MyComposeView.kt (hunk 5 above)
Comment (erksch) [reply example/android/shared/src/commonMain/kotlin/com/myrnproject/shared/MyComposeView.kt:44]: data classes and multiple parameters done. return types and exceptions are not possible as these are events

**Pull request merged**